sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbird/integration/service/SunbirdAuthenticationService.java

```diff
@@ -0,0 +1,145 @@
+            kycExchangeRequestDto.setClaimLocales(Arrays.asList(kycExchangeDto.getClaimsLocales()));
+
+            //set signature header, body and invoke kyc exchange endpoint
+            String requestBody = objectMapper.writeValueAsString(kycExchangeRequestDto);
+            RequestEntity requestEntity = RequestEntity
+                    .post(UriComponentsBuilder.fromUriString(kycExchangeUrl).pathSegment(relyingPartyId,
+                            clientId).build().toUri())
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .body(requestBody);
+            ResponseEntity<ResponseWrapper<KycExchangeResponseDto>> responseEntity = restTemplate.exchange(requestEntity,
+                    new ParameterizedTypeReference<>() {
+                    });
+
+            if (responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
+                ResponseWrapper<KycExchangeResponseDto> responseWrapper = responseEntity.getBody();
+                if (responseWrapper.getResponse() != null && responseWrapper.getResponse().getKyc() != null) {
+                    return new KycExchangeResult(responseWrapper.getResponse().getKyc());
+                }
+                log.error("Errors in response received from IDA Kyc Exchange: {}", responseWrapper.getErrors());
+                throw new KycExchangeException(CollectionUtils.isEmpty(responseWrapper.getErrors()) ?
+                        ErrorConstants.DATA_EXCHANGE_FAILED : responseWrapper.getErrors().get(0).getErrorCode());
+            }
+
+            log.error("Error response received from IDA (Kyc-exchange) with status : {}", responseEntity.getStatusCode());
+        } catch (KycExchangeException e) {
+            throw e;
+        } catch (Exception e) {
+            log.error("IDA Kyc-exchange failed with clientId : {}", clientId, e);
+        }
+        throw new KycExchangeException("mock-ida-005", "Failed to build kyc data");
+    }
+
+    @Override
+    public SendOtpResult sendOtp(String relyingPartyId, String clientId, SendOtpDto sendOtpDto)
+            throws SendOtpException {
+        return null;
+        }
+
+    @Override
+    public boolean isSupportedOtpChannel(String channel) {
+        return authenticationHelperService.isSupportedOtpChannel(channel);
+    }
+
+    @Override
+    public List<KycSigningCertificateData> getAllKycSigningCertificates() {
+        List<KycSigningCertificateData> certs = new ArrayList<>();
+        AllCertificatesDataResponseDto allCertificatesDataResponseDto = keymanagerService.getAllCertificates(APPLICATION_ID,
+                Optional.empty());
+        for (CertificateDataResponseDto dto : allCertificatesDataResponseDto.getAllCertificates()) {
+            certs.add(new KycSigningCertificateData(dto.getKeyId(), dto.getCertificateData(),
+                    dto.getExpiryAt(), dto.getIssuedAt()));
+        }
+        return certs;
+    }
+}
```

Comment:
Unit test case file required for this class with test case for each condition handlings.

Reply:
If you want to do this later, can you create a sub task in JIRA, so we don't miss it ?

Reply:
added a subtask for this

Reply:
can you provide the sub task link here for other to follow ?

Reply:
https://mosip.atlassian.net/browse/ES-622